dev-lang/zig/zig-0.4.0.ebuild

```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit cmake-utils
+
+DESCRIPTION="A robust, optimal, and maintainable programming language."
+HOMEPAGE="https://ziglang.org/"
+LICENSE="MIT"
+SLOT="0"
+IUSE=""
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://github.com/ziglang/zig.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/ziglang/zig/archive/${PV}.tar.gz -> zig-${PV}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+ALL_LLVM_TARGETS=( AArch64 AMDGPU ARM BPF Hexagon Lanai Mips MSP430 NVPTX
+	PowerPC Sparc SystemZ WebAssembly X86 XCore )
+ALL_LLVM_TARGETS=( "${ALL_LLVM_TARGETS[@]/#/llvm_targets_}" )
+# According to zig's author, zig builds that do not support all targets are not
+# supported by the upstream project.
+LLVM_TARGET_USEDEPS=${ALL_LLVM_TARGETS[@]}
+
+RDEPEND="~sys-devel/llvm-8.0.0:=[${LLVM_TARGET_USEDEPS// /,}]"
+DEPEND="
+	${RDEPEND}
+	dev-util/cmake
+	|| (
```

Comment:
This isn't the correct way to check if gcc or clang requirements are satisfied. I wouldnt worry about this honestly, as these versions are so old default profiles should already depend on newer ones. I dont think there even is clang-3 in the tree anymore. 

However if you truly wish to make these checks, use toolchain-funcs.eclass and pkg_setup phase instead.